### PR TITLE
Triangular matrix multiplication

### PR DESCRIPTION
**Pull request by Sely85**
Local version (8 cases) of the triangular matrix multiplication algorithm (TRMM).

### CODE REVIEW

**Comment by rasolca** [comment]
bors try

**Comment by rasolca** [comment]
bors try

**Review by albestro — CHANGES_REQUESTED** [review]

File: include/dlaf/multiplication/triangular/impl.h [hunk 1]
```diff
@@ -0,0 +1,399 @@
...
+  auto executor_np = dlaf::getNpExecutor<backend>();
+  SizeType m = mat_b.nrTiles().rows();
+  SizeType n = mat_b.nrTiles().cols();
+
+  for (SizeType k = m - 1; k >= 0; --k) {
+    for (SizeType j = 0; j < n; ++j) {
+      auto kj = LocalTileIndex{k, j};
+
+      for (SizeType i = k + 1; i < m; ++i) {
+        auto& trailing_executor = (i == k + 1) ? executor_hp : executor_np;
```
Comment (rasolca) [inline include/dlaf/multiplication/triangular/impl.h:212]: The TRMM DAG doesn't need any gemm to be HP. You can schedule them all as NP tasks. (It applies for all 8 cases).

File: include/dlaf/multiplication/triangular.h [hunk 2]
```diff
@@ -0,0 +1,90 @@
...
+/// @param side specifies whether op(A) appears on the \a Left or on the \a Right of matrix B,
+/// @param uplo specifies whether the matrix A is a \a Lower or an \a Upper triangular matrix,
+/// @param op specifies the form of op(A) to be used in the matrix multiplication: \a NoTrans, \a Trans,
+/// \a ConjTrans,
+/// @param diag specifies if the matrix A is assumed to be unit triangular (\a Unit) or not (\a NonUnit),
+/// @param mat_a contains the triangular matrix A. Only the tiles of the matrix which contain the upper or
+/// the lower triangular part (depending on the value of uplo) are accessed in read-only mode (the
+/// elements are not modified),
+/// @param mat_b on entry it contains the matrix B, on exit the matrix elements are overwritten with the
+/// elements of the transformed matrix.
```
Comment (rasolca) [inline include/dlaf/multiplication/triangular.h:34]: Just a last minor thing:
```suggestion
/// elements of the result.
```

**Review by albestro — APPROVED** [review]

**Comment by rasolca** [comment]
bors try

**Review by rasolca — APPROVED** [review]

**Pull request merged**